templates/todo/projects/csharp-sql/.repo/bicep/infra/main.bicep

```diff
@@ -54,139 +55,266 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+      }
+      {
+        objectId: api.outputs.systemAssignedMIPrincipalId
+        permissions: {
+          secrets: [ 'get', 'list' ]
+        }
+      }
+    ]
+    secrets:{
+      secureList: [
+        {
+          name: 'sqlAdmin'
+          value: sqlAdminPassword
+        }
+        {
+          name: 'appUser'
+          value: appUserPassword 
+        }
+        {
+          name: connectionStringKey
+          value: 'Server=${sqlService.outputs.name}${environment().suffixes.sqlServerHostname}; Database=${!empty(sqlDatabaseName) ? sqlDatabaseName : 'Todo'}; User=${appUser}; Password=${appUserPassword}'
+        }
+      ]
+    }
   }
 }
 
 // The application database
-module sqlServer '../../../../../common/infra/bicep/app/sqlserver.bicep' = {
-  name: 'sql'
+module sqlService 'br/public:avm/res/sql/server:0.2.0' = {
+  name: 'sqlservice'
   scope: rg
   params: {
     name: !empty(sqlServerName) ? sqlServerName : '${abbrs.sqlServers}${resourceToken}'
-    databaseName: sqlDatabaseName
+    administratorLogin: sqlAdmin
+    administratorLoginPassword: sqlAdminPassword
     location: location
     tags: tags
-    sqlAdminPassword: sqlAdminPassword
+    publicNetworkAccess: 'Enabled'
+    databases: [
+      {
+        name: !empty(sqlDatabaseName) ? sqlDatabaseName : 'Todo'
+      }
+    ]
+    firewallRules:[
+      {
+        name: 'Azure Services'
+        startIpAddress: '0.0.0.1'
+        endIpAddress: '255.255.255.254'
+      }
+    ]
+  }
+}
+
+module sqldeploymentscript '../../../../../common/infra/bicep/app/sql-deployment-script.bicep' = {
+  scope: rg
+  name: 'sqldeploymentscript'
+  params: {
+    location: location
     appUserPassword: appUserPassword
-    keyVaultName: keyVault.outputs.name
+    sqlAdminPassword: sqlAdminPassword
+    sqlDatabaseName: !empty(sqlDatabaseName) ? sqlDatabaseName : 'Todo'
+    sqlServiceName: sqlService.outputs.name
   }
 }
 
 // Create an App Service Plan to group applications under the same payment plan and SKU
-module appServicePlan '../../../../../../common/infra/bicep/core/host/appserviceplan.bicep' = {
+module appServicePlan 'br/public:avm/res/web/serverfarm:0.1.0' = {
   name: 'appserviceplan'
   scope: rg
   params: {
     name: !empty(appServicePlanName) ? appServicePlanName : '${abbrs.webServerFarms}${resourceToken}'
-    location: location
-    tags: tags
     sku: {
+      capacity: 1
+      family: 'B'
       name: 'B1'
+      size: 'B1'
+      tier: 'Basic'
     }
+    location: location
+    tags: tags
+    reserved: true
+    kind: 'Linux'
   }
 }
 
-// Store secrets in a keyvault
-module keyVault '../../../../../../common/infra/bicep/core/security/keyvault.bicep' = {
+// // Store secrets in a keyvault
+module keyVault 'br/public:avm/res/key-vault/vault:0.3.5' = {
   name: 'keyvault'
   scope: rg
   params: {
     name: !empty(keyVaultName) ? keyVaultName : '${abbrs.keyVaultVaults}${resourceToken}'
     location: location
     tags: tags
-    principalId: principalId
```

Comment:
How will user be able to run the app locally without this?

Reply:
I added permissions elsewhere instead, the link is here: https://github.com/Azure/azure-dev/pull/3620/files/a35fc18c3ab15d73fc88c49fd0ccb75fc029eb82#diff-f1fba4c513e3fd25a2a0c835cbfe410e23057d6680763ac3f0bd1ebdf630b8abR132.
If you want to add the permission of `principalId` at the current location, it is also possible. We can modify the relevant code to complete it.

Reply:
Ok, as-is is fine